integration/spark/app/src/test/java/io/openlineage/spark/agent/lifecycle/LibraryTest.java

```diff
@@ -173,12 +171,37 @@ void testRDDName(SparkSession spark) {
     assertThat(s).isEqualTo("map_partitions_numbers");
   }
 
+  @SneakyThrows
   private void verifySerialization(List<OpenLineage.RunEvent> events)
       throws JsonProcessingException {
     for (OpenLineage.RunEvent event : events) {
-      assertNotNull(
-          "Event can serialize",
-          OpenLineageClientUtils.newObjectMapper().writeValueAsString(event));
+      assertNotNull("Event can serialize", writeValueAsString(event));
     }
   }
+
+  // workaround to use OpenLineage client's ObjectMapper which is shaded
+  private Map<String, Object> readValue(File src)
```

Comment:
i don't like the fact that those 3 methods are almost identical, but I don't see any good solution to extract some generic part

Reply:
Please look at my comment: https://github.com/OpenLineage/OpenLineage/pull/1650/files#r1118760545
This at least describes why is that.

Reply:
I've managed to get rid of those.